aws/resource_aws_rds_cluster.go

```diff
@@ -650,6 +697,109 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 			return err
 		}
 
+	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok {
+		pointInTime := v.([]interface{})[0].(map[string]interface{})
+		createOpts := &rds.RestoreDBClusterToPointInTimeInput{
+			DBClusterIdentifier:       aws.String(identifier),
+			DeletionProtection:        aws.Bool(d.Get("deletion_protection").(bool)),
+			SourceDBClusterIdentifier: aws.String(pointInTime["source_cluster_identifier"].(string)),
+			Tags:                      tags,
+		}
+
+		if v, ok := pointInTime["restore_to_time"].(string); ok {
+			restoreToTime, _ := time.Parse(time.RFC3339, v)
+			createOpts.RestoreToTime = aws.Time(restoreToTime)
+		}
+
+		if v, ok := pointInTime["use_latest_restorable_time"].(bool); ok && v {
+			createOpts.UseLatestRestorableTime = aws.Bool(v)
+		}
+
+		if createOpts.RestoreToTime == nil && createOpts.UseLatestRestorableTime == nil {
+			return fmt.Errorf(`provider.aws: aws_rds_cluster: %s: Either "restore_to_time" or "use_latest_restorable_time" must be set`, d.Get("database_name").(string))
+		}
+
+		if attr, ok := pointInTime["restore_type"].(string); ok {
+			createOpts.RestoreType = aws.String(attr)
+		}
+
+		if err := validateRestoreToPointInTimeInput(d); err != nil {
+			return err
+		}
```

Comment:
looking this over in more detail, I think this validation might hinder operators' experiences as the values checked w/in this method can technically be added without any issue to the cluster creation, given they match the original cluster's values i.e.
```
resource "aws_rds_cluster" "restored_pit" {
  cluster_identifier  = "%s"
  skip_final_snapshot = true
  engine              = aws_rds_cluster.test.engine
  database_name       = aws_rds_cluster.test.database_name
  master_username     = aws_rds_cluster.test.master_username
  storage_encrypted   = aws_rds_cluster.test.storage_encrypted
  source_region       = aws_rds_cluster.test.source_region
  restore_to_point_in_time {
    source_cluster_identifier  = aws_rds_cluster.test.cluster_identifier
    restore_type               = "full-copy"
    use_latest_restorable_time = true
  }
}
```
To account for the possible case where values provided do not match the original cluster, resulting in a perpetual diff after `apply`, I would suggest documenting the expected behavior in the docs